src/partials/messenger.ts

```diff
@@ -557,9 +588,86 @@ class ConversationController {
+            this.currentMentionFilterWord = currentWord.substr(1);
+            let query = this.currentMentionFilterWord.toLowerCase().trim();
+            const selectedMentionObject = this.getSelectedMention();
+            this.currentMentions = this.allMentions.filter((i) => {
+                if (query.length === 0) {
+                    return true;
+                }
+                return i.query.indexOf(query) >= 0;
+            });
+            // If only one mention is filtered, select them
+            if (this.currentMentions.length === 1) {
+                this.selectedMention = 0;
+            } else if (selectedMentionObject !== null) {
+                // Get the new position of the latest selected mention object
+                this.selectedMention = null;
+                this.selectedMention = this.currentMentions.findIndex((m) => {
+                    return m.identity === selectedMentionObject.identity;
+                });
+            }
+        } else {
+            this.currentMentionFilterWord = null;
+        }
+    }
+
+    public getSelectedMention = (): {identity: string, query: string, isAll: boolean} => {
+        if (this.selectedMention === null
+            || this.selectedMention < 0
+            || this.selectedMention > this.currentMentions.length - 1) {
+            return null;
+        }
+
+        return this.currentMentions[this.selectedMention];
+    }
+    /**
+     * Handle mention selector navigation
+     */
+    public onComposeKeyDown = (ev: KeyboardEvent): boolean => {
+        if (this.showMentionSelector() && !ev.shiftKey) {
+            let move = ev.which === 40 ? 1 : (ev.which === 38 ? - 1 : 0);
+            if (move !== 0) {
+                // Move cursors position in mention selector
+                if (this.selectedMention !== null) {
+                    this.selectedMention += move;
+                    // Fix positions
+                    if (this.selectedMention > this.currentMentions.length - 1) {
+                        this.selectedMention = 0;
+                    } else if (this.selectedMention < 0) {
+                        this.selectedMention = this.currentMentions.length - 1;
+                    }
+                } else {
+                    this.selectedMention = 0;
+                }
+                return false;
+            }
+
+            if (ev.which === 13) {
+                // Enter, select current mention
+                const selectedMentionObject = this.getSelectedMention();
+                if (selectedMentionObject === null) {
+                    // If no (or a invalid) mention is selected, select the first mention
+                    this.selectedMention = 0;
+                } else {
+                    this.onMentioned(selectedMentionObject.identity);
+                }
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    public onMentioned(identity: string = null): void {
```

Comment:
Maybe we should rename this to `onMentionSelected` to prevent confusion with mentions of yourself? 🙂